src/Supervisor.cpp

```diff
@@ -1221,4 +1221,60 @@ ZunResult Supervisor::FadeOutMusic(f32 fadeOutSeconds)
     return ZUN_SUCCESS;
 }
 #pragma optimize("", on)
+
+#pragma optimize("", on)
+// this is for rebinding keys
+u8 *th06::Controller::GetControllerState()
+{
+    static u32 controllerData[32];
+    memset(&controllerData, 0, sizeof(controllerData));
+    if (g_Supervisor.controller == NULL)
+    {
+        // TODO: not tested
+        JOYINFOEX joyinfo;
```

Comment:
Please put all the variable definitions at the start. It makes it easier to deal with the stack slots.